phantomgamesTests/SearchGameViewModelTests.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  SearchViewModelTests.swift
+//  phantomgamesTests
+//
+//  Created by Aphiwe Shozi on 2024/05/06.
+//
+
+import XCTest
+@testable import phantomgames
+
+final class SearchGameViewModelTests: XCTestCase {
+    
+    var viewModel: SearchGameViewModel!
+    var mockRepository: MockSearchGameRepository!
+    var mockDelegate: MockViewModelDelegate!
+    
+    
+    override func setUp() {
+        mockRepository = MockSearchGameRepository()
+        mockDelegate = MockViewModelDelegate()
+        viewModel = SearchGameViewModel(repository: mockRepository, delegate: mockDelegate)
+    }
+    class MockSearchGameRepository: SearchGameRepositoryType {
+        var shouldReturnError = false
+        let mockGameList = [
+            Game(id: 1, title: "game1", thumbnail: "Thumbnail1", shortDescription: "shortDes1", gameURL: "URL1", genre: "genre1", platform: "platform1", publisher: "publisher1", developer: "dev1", releaseDate: "Date1", freetogameProfileURL: "Profile1"),
+            Game(id: 2, title: "game2", thumbnail: "Thumbnail2", shortDescription: "shortDes2", gameURL: "URL2", genre: "genre2", platform: "platform2", publisher: "publisher2", developer: "dev2", releaseDate: "Date2", freetogameProfileURL: "Profile2")
+        ]
+        
+        func fetchSearchResults(completion: @escaping (phantomgames.SearchGameResult)) {
+            if shouldReturnError {
+                completion(.failure(APIError.serverError))
+            }
+            else {
+                completion(Result.success(mockGameList))
+            }
+        }
+    }
+    
+    class MockViewModelDelegate: SearchGameViewModelDelegate {
+        var showErrorCalled = false
+        var errorMessage: String?
+        
+        func reloadView() {}
+        
+        func show(error: String) {
+            showErrorCalled = true
+            errorMessage = error
+        }
+    }
+    
+    func testGamelistCount () {
+        viewModel.fetchSearchResults()
+        let result = viewModel.gameListCount
+        XCTAssertEqual(result, 2)
+    }
+    
+    func testGameAtIndex () {
+        viewModel.fetchSearchResults()
+        
+        let expectedResult = Game(id: 1, title: "game1", thumbnail: "Thumbnail1", shortDescription: "shortDes1", gameURL: "URL1", genre: "genre1", platform: "platform1", publisher: "publisher1", developer: "dev1", releaseDate: "Date1", freetogameProfileURL: "Profile1").title
+        let result = viewModel.game(atIndex: 0)?.title
+        XCTAssertNotNil(result)
+        XCTAssertEqual(result, expectedResult)
```

Comment:
Rather remove Expected result var. It is not necessary

```suggestion
        XCTAssertEqual(result, "game1")
```